provers/sp1/driver/src/lib.rs

```diff
@@ -109,6 +109,12 @@ impl Prover for Sp1Prover {
         let param = Sp1Param::deserialize(config.get("sp1").unwrap()).unwrap();
         let mode = param.prover.clone().unwrap_or_else(get_env_mock);
 
+        if matches!(param.recursion, RecursionMode::Compressed) {
+            return Err(ProverError::GuestError(
```

Comment:
I think you can not do that before we have a clear flag shows it is aggregation.